salt/grains/core.py

```diff
@@ -1688,6 +1688,8 @@ def ip_fqdn():
                 info = socket.getaddrinfo(_fqdn, None, socket_type)
                 ret[key] = list(set(item[4][0] for item in info))
             except socket.error:
+                log.warning('Unable to find IPv%s record for "%s". '
+                            'Create a dns record or add to /etc/hosts to clear this message.', ipv_num, _fqdn)
```

Comment:
This is a fine approach but let's change this error message to indicate that failure to address this might have performance implications.